govc/vm/customize.go

```diff
@@ -246,6 +250,18 @@ func (cmd *customize) Run(ctx context.Context, f *flag.FlagSet) error {
 		}
 	}
 
+	if cmd.username != "" {
```

Comment:
This check and the one below isn't needed, since the UserData fields being set are the same type (`string`) and same default value (`""`) as the cmd flags. Looking at the existing code, there's another check than can be removed, you could then add the new cmd fields there:
```patch
diff --git a/govc/vm/customize.go b/govc/vm/customize.go
index 356ab6f2..82762486 100644
--- a/govc/vm/customize.go
+++ b/govc/vm/customize.go
@@ -211,12 +211,12 @@ func (cmd *customize) Run(ctx context.Context, f *flag.FlagSet) error {
 	sysprep, isWindows := spec.Identity.(*types.CustomizationSysprep)
 	linprep, _ := spec.Identity.(*types.CustomizationLinuxPrep)
 
-	if cmd.domain != "" {
-		if isWindows {
-			sysprep.Identification.JoinDomain = cmd.domain
-		} else {
-			linprep.Domain = cmd.domain
-		}
+	if isWindows {
+		sysprep.Identification.JoinDomain = cmd.domain
+		// add username/org fields below
+		sysprep.UserData.{...} = ...
+	} else {
+		linprep.Domain = cmd.domain
 	}
 
 	if len(cmd.dnsserver) != 0 {

```